pkg/argocd/argocd.go

```diff
@@ -675,17 +675,34 @@ func getApplicationSourceType(app *v1alpha1.Application) v1alpha1.ApplicationSou
 	return app.Status.SourceType
 }
 
-// getApplicationSource returns the main source of a Helm or Kustomize type of the application
+// getApplicationSource returns the source matching git-repository annotation
+// or the Helm/Kustomize source, or the primary source if no match is found
 func getApplicationSource(app *v1alpha1.Application) *v1alpha1.ApplicationSource {
```

Comment:
Changing this func will impact many places, and in some of these places, we do want to get the primary source, not the one for write-back. I think we should avoid changing this func, and instead modify git.go